blocks/api/serializer.js

```diff
@@ -115,6 +115,10 @@ export function serializeBlock( block ) {
 	const saveContent = getSaveContent( blockType, block.attributes );
 	const saveAttributes = getCommentAttributes( block.attributes, parseBlockAttributes( saveContent, blockType ) );
 
+	if ( 'wp:core/more' === blockName ) {
+		return `<!-- more ${ saveAttributes.customText ? `${ saveAttributes.customText } ` : '' }-->${ saveAttributes.noTeaser ? '\n<!--noteaser-->' : '' }`;
```

Comment:
This should actually be `<!--more` instead of `<!-- more` according to `get_the_content()`.